### PR TITLE
[1.9.4] Added CriticalHitEvent

### LINKED ISSUE #2871: [Suggestion] Event to handle critical hits
In 1.8.9 it is not possible to determin if a player make a critical hit nor prevent him form doing one. An external event for handling it whould be very nice. We could also add an event for handling critical enchantment hits.

If I am allowed, then I can make a pull request. #

### PR DESCRIPTION
**Pull request by mcenderdragon**
Reopened #2883 (I messed up git).
Also doing some improves.
Fixes #2871

### CODE REVIEW

**Comment by mcenderdragon** [comment]
Added an example https://gist.github.com/mcenderdragon/b2e11c054dca08c566934a860087d881

File: src/main/java/net/minecraftforge/event/entity/player/CriticalHitEvent.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        this.damageModifier = mod;
+    }
+    
+    public float getDamageModifier()
+    {
+        return this.damageModifier;
+    }
+	
+	public EntityLivingBase getTarget()
+    {
```
Comment (iLexiconn) [inline src/main/java/net/minecraftforge/event/entity/player/CriticalHitEvent.java]: Formatting

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Comment by mcenderdragon** [comment]
this is targetted to the 1.9.4 branch, but this has allready be merged with the master. Is there a way to retarged a pull request ? (without recreating it)

**Comment by AEnterprise** [comment]
there is not, you will have to close this one and make a new one

**Comment by mcenderdragon** [comment]
this will be then the 3rd try... :/

**Pull request closed without merging**